### ISSUE TITLE
Create Documentation for Test Curation Guidelines 

### ISSUE THREAD

**Issue by nomadtechie** [tc39/test262#1598]
See #1597 for the background. 

I'll like to suggest that we create a `Test Curation Guide` to capture the intended workflow for moving tests out of `implementation-contributed/**/` directories and into `test262/tests/`. 

For instance, a curation process for may involve this workflow: 

Identify is a test file is:
- valid and correct
- new coverage or duplicate coverage
- needs rework to be separated into multiple test files
- needs refactoring or cleanup

After this curation, some of the outcomes would be:
- refactoring the test to use test262 frontmatter tools and dependencies and moving the test to the appropriate directory in `test262/tests/`
-  deleting a test bc its duplicate coverage
- filling a bug or patch in for implementation project to fix, or remove tests

On the other hand, a curation process should _*not involve the following*_: 
- adding new files in the `implementation-contributed/**/` directories
- renaming, moving, restructuring files, file extensions, or directories in `implementation-contributed/**/`

As part of the automation work highlighted in #1597, I'm creating a resolver algorithm to manage changes in both target and source directories. (target=`test262/implementation-contributed/**/`, source=`implementation/tests`). 

The rules dictated in that algorithm should be reflective of the guidelines so `git` nightmares and unexpected changes can be avoided. 

In summary, for the curation process, deleting a file, and in place file modications are expected, and new file addtions, moves/renames, and changes to file extensions would be unexpected changes.

**Comment by rwaldron** [comment]
> In summary, for the curation process, deleting a file, and in place file modifications are expected, 

Confirmed, but we reserve the authority to disallow "in place file modifications" at any time in the future. 

> and new file addtions, moves/renames, and changes to file extensions would be unexpected changes.

Confirmed, and dare I say: strictly forbidden.

**Comment by nomadtechie** [comment]
Sounds good Rick! Thanks for confirming.
